File: AndroidArchitecture/app/src/main/java/tisoul/dev/androidarchitecture/mvp/MVPProfileFragment.java
package tisoul.dev.androidarchitecture.mvp;


import android.app.Fragment;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;

import javax.inject.Inject;

import tisoul.dev.androidarchitecture.ProfileActivity;
import tisoul.dev.androidarchitecture.R;

public class MVPProfileFragment
        extends Fragment
        implements Profile.View {

    EditText tvName;
    EditText tvEmail;

    @Inject
    Profile.Presenter presenter;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setUpInjector();

        presenter.setView(this);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_profile_mvp, container, false);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        setUpView(view);
    }

    @Override
    public void onStart() {
        super.onStart();

        presenter.subscribe();
    }

    @Override
    public void onStop() {
        super.onStop();

        presenter.unSubscribe();
    }

    @Override
    public void setName(CharSequence name) {
        tvName.setText(name);
    }

    @Override
    public void setEmail(CharSequence email) {
        tvEmail.setText(email);
    }

    public void onUpdateClick(View view) {
        presenter
                .update(
                        tvName.getText(),
                        tvEmail.getText());
    }

    private void setUpView(View view) {
        tvName = (EditText) view.findViewById(R.id.tv_name);
        tvEmail = (EditText) view.findViewById(R.id.tv_email);

        view.findViewById(R.id.bt_update)
                .setOnClickListener(this::onUpdateClick);
    }

    private void setUpInjector() {
        ((ProfileActivity) getActivity())
                .profileComponent
                .inject(this);
    }
}

File: AndroidArchitecture/app/src/main/java/tisoul/dev/androidarchitecture/mvp/MVPProfileFragment.kt
package tisoul.dev.androidarchitecture.mvp


import android.app.Fragment
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.EditText
import tisoul.dev.androidarchitecture.ProfileActivity
import tisoul.dev.androidarchitecture.R
import tisoul.dev.androidarchitecture.commons.inflate
import javax.inject.Inject

class MVPProfileFragment : Fragment()
        , Profile.View {

    internal var tvName: EditText? = null
    internal var tvEmail: EditText? = null

    @Inject
    lateinit var presenter: Profile.Presenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setUpInjector()

        presenter.setView(this)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return container?.inflate(R.layout.fragment_profile_mvp)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        setUpView(view)
    }

    override fun onStart() {
        super.onStart()

        presenter.subscribe()
    }

    override fun onStop() {
        super.onStop()

        presenter.unSubscribe()
    }

    override fun setName(name: CharSequence) {
        tvName?.setText(name)
    }

    override fun setEmail(email: CharSequence) {
        tvEmail?.setText(email)
    }

    fun onUpdateClick() {
        presenter
                .update(
                        tvName?.text!!.toString(),
                        tvEmail?.text!!.toString())
    }

    private fun setUpView(view: View) {
        tvName = view.findViewById(R.id.tv_name) as EditText
        tvEmail = view.findViewById(R.id.tv_email) as EditText

        view.findViewById(R.id.bt_update)
                .setOnClickListener({ onUpdateClick() })
    }

    private fun setUpInjector() {
        (activity as ProfileActivity)
                .profileComponent!!
                .inject(this)
    }
}
